### DISCUSSION TITLE
Using coupled PDEs in PINN

### DISCUSSION THREAD

**Discussion by pranavps-creator** [lululxvi/deepxde#1044] [Q&A] [answered]
Hi Lulu, 

I am trying to solve a simple PDE using DeepXDE. I always run into some compilation error while running the code that I have written. I have referred to the discussion- [](https://github.com/lululxvi/deepxde/issues/31) and I have created two seperated functions func1 and func2 after referring to this discussion. Below are the equations and my code- 

Equation- 

![image](https://user-images.githubusercontent.com/54719978/203591869-bd1f3edb-46e9-44a8-b3cd-118bc371f24b.png)

Here's my code- 
```
import deepxde as dde
import numpy as np
from deepxde.backend import tf

def pde(t,x): # defining residual of the PDEs

    x1,x2 = x[:,0:1],x[:,1:2]
    dx_dt= dde.grad.Jacobian(x1, t, i=0, j=0)
    dy_dt= dde.grad.Jacobian(x2, t, i=0, j=0)
    return(dx_dt-4*x1+2*x2, dy_dt+x2-3*x)

def func1(t):
   return(3*tf.exp(t[:,0:1])-2*tf.exp(t[:,0:1]))

def func2(t):
 return(2*tf.exp(t[:,0:1])-tf.exp(2*t[:,0:1]))

geom = dde.geometry. geometry_2d. Rectangle([0,0],[2,2])
timedomain = dde.geometry.TimeDomain(0, 1)
geomtime = dde.geometry.GeometryXTime(geom, timedomain) # define geometric and time domains

ic =[0]*2
ic[0]=dde.icbc.initial_conditions.IC(geomtime, func1 , lambda _, on_initial: on_initial)
ic[1]=dde.icbc.initial_conditions.IC(geomtime, func2, lambda _, on_initial: on_initial)

data = dde.data.TimePDE(geomtime,pde,[ic],num_domain=320, solution=[func1,func2], num_test=80000)

layer_size = [2] + [30] * 6 + [2]
activation = "tanh"
initializer = "Glorot uniform" 
net = dde.nn.FNN(layer_size, activation, initializer)

model = dde.Model(data, net)
model.compile("adam", lr=1e-3, metrics=["l2 relative error"])
losshistory, train_state = model.train(iterations=20000)
'''

**Comment by elliot-hicks** [comment]
Hi @pranavps-creator, generally its easier to read if you put the entire notebook in the code format so that the whole thing is visible in one cell - this is hard to read for those trying to help you! Also an example of the error message would be useful for diagnosing your issue.

As a sanity check, can you change your PDE return from `return(dx_dt-4*x1+2*x2, dy_dt+x2-3*x)` to `return [dx_dt-4*x1+2*x2, dy_dt+x2-3*x]` ?

**Reply by pranavps-creator** [reply]
Hi @elliot-hicks, Thank you for the feedback. I have put my entire code in one cell. I have also tried changing the return function. Here is the error that I get- 

dx_dt= dde.grad.Jacobian(x1, t, i=0,j=0)
TypeError: Jacobian.__init__() got an unexpected keyword argument 'i'

**Reply by elliot-hicks** [reply, marked as answer]
Hey @pranavps-creator ,

No problem here, I beleive youre accidentally initialising a class (**J**acobian) whereas you want **j**acobian. 

Change all `dde.grad.Jacobian(...)` to `dde.grad.jacobian(...)`. Its easy to get them mixed up like this. Take a look at the source code here and you'll see the mistake: 
https://deepxde.readthedocs.io/en/latest/_modules/deepxde/gradients.html#jacobian

**Reply by pranavps-creator** [reply]
Thanks, @elliot-hicks . This worked well for me
